### DISCUSSION TITLE
First preview version of the updated Grimoire is here!

### DISCUSSION THREAD

**Discussion by goniszewski** [goniszewski/grimoire#115] [Announcements]
It is my pleasure to announce the release of Grimoire _v0.4.0-pre.1_, a preview of our extensively refactored project. This version brings significant improvements and simplifications:

🚀 Streamlined Setup: Now runs in a single container - no more separate PocketBase container!
⚡ Enhanced Performance: We've migrated to [Bun](https://bun.sh) for faster operations and are connecting to the database directly with [Drizzle ORM](https://orm.drizzle.team).
🛠️ Improved Maintainability: The new architecture is easier to maintain, extend and deploy.

Try it out now:
`docker pull goniszewski/grimoire:develop`

I can't wait to hear your feedback on this significant step for Grimoire. Give it a spin and share with us know what you think!

Link to preview release: [https://github.com/goniszewski/grimoire/releases/tag/v0.4.0-pre.1](https://github.com/goniszewski/grimoire/releases/tag/v0.4.0-pre.1)

Happy bookmarking! 📚✨

**Comment by snowy-jaguar** [comment]
I appear to be getting "Internal Server Error" when I try to deploy the stack as is (non preview version, I think). Is it worth me doing that or would it be better if I just waited for the full release of the new version?

**Reply by goniszewski** [reply]
Hi @SnowyJaguar1034! I'd suggest starting with the preview version and then switching to the stable release later. The updated Grimoire is pretty stable and works great with the available browser extension.

**Reply by snowy-jaguar** [reply]
@goniszewski Thanks for the reply! I'll give the preview/pre-release version a go and get back to you on if I have any issues.

I presume the branch with the pre-release version is "develop"? It's currently pinned to the "latest" container image and not the "develop" one you mention in pre-release announcement 1.

**Reply by goniszewski** [reply]
Yeah, the README.md could be updated to reflect all the changes. Be sure to use the [develop](https://hub.docker.com/layers/goniszewski/grimoire/develop/images/sha256-6bda197b04d9484fc930ea2ccbc33ad0c1faf45757f98b09ef4a1df8cd883641?context=explore) tag for the Docker image!

**Reply by snowy-jaguar** [reply]
> Yeah, the README.md could be updated to reflect all the changes. Be sure to use the [develop](https://hub.docker.com/layers/goniszewski/grimoire/develop/images/sha256-6bda197b04d9484fc930ea2ccbc33ad0c1faf45757f98b09ef4a1df8cd883641?context=explore) tag for the Docker image!

I gave the `develop` tag a whirl earlier today and received a 501 (or 50x) internal error code. I was trying it with some minor adjustments to make it work with portainer so I'm going to try the bone stock version tomorrow and see what I get. I'll then try troubleshooting why the error occured. I might pr the changes as they're pretty minor and should apply to any other container orchestration platform like portainer, dockge, rancher (kubernetes).

**Reply by snowy-jaguar** [reply]
sorry I parked this for a little while, I'll get back to it soon I hope as your project looks really good and will hopefully be able to replace linkwarden

**Reply by goniszewski** [reply]
Thanks @SnowyJaguar1034, I appreciate that!

Why do you want to move from Linkwarden? Just curious 🙂

**Reply by snowy-jaguar** [reply]
A few reasons:
- Your project appears to grab more metadata about a page than Linkwardens does.
- Your project (from the screenshots) looks like a very nice UI
- I'm always interested in trying alternatives. I was never particularly sold on Linkwarden, but I spun it up as I needed something and it was easy enough to get set up - until it decided to wipe itself one day forcing me to recover the entire host from a backup purely for linkwarden.
- I was hoping your project could auto-generate tags for me (I now think that's not a feature) as I'm super lazy when it comes to tagging stuff but they'd probably help me out.
